workflow-controller/controller-service/src/main/java/io/hyscale/controller/profile/ProfileSpecProcessor.java

```diff
@@ -0,0 +1,119 @@
+
+    public HyscaleInputSpec process(ProfileArg profileArg, List<File> serviceSpecFiles) throws HyscaleException {
+        List<File> profiles = null;
+        boolean strict = false;
+        if (profileArg != null) {
+            if (StringUtils.isNotBlank(profileArg.getProfileName())) {
+                profiles = profileLocator.locateProfiles(profileArg.getProfileName(), serviceSpecFiles);
+                strict = true;
+            } else {
+                profiles = profileArg.getProfiles();
+            }
+        }
+        if (!profileSpecInputValidator.validate(profiles)) {
+            throw new HyscaleException(ControllerErrorCodes.PROFILE_VALIDATION_FAILED);
+        }
+        checkForServiceDependency(serviceSpecFiles, profiles, strict);
+        HyscaleInputSpec hyscaleInputSpec = new HyscaleInputSpec();
+        hyscaleInputSpec.setServiceSpecFiles(serviceSpecFiles);
+        hyscaleInputSpec.setProfileFiles(profiles);
+        return hyscaleInputSpec;
+    }
+
+    private void checkForServiceDependency(List<File> serviceSpecFiles,
+                                           List<File> profileFiles, boolean strictProfile) throws HyscaleException {
+        Set<String> serviceFromSpec = new HashSet<>();
+        Set<String> serviceFromProfiles = new HashSet<String>();
+
+        serviceFromSpec = serviceSpecFiles.stream().map(eachFile -> {
+            try {
+                return ServiceSpecUtil.getServiceName(eachFile);
+            } catch (HyscaleException e) {
+                return null;
+            }
+        }).collect(toSet());
+
+        serviceFromProfiles = profileFiles.stream().map(eachFile -> {
+            try {
+                return ServiceProfileUtil.getServiceNameFromProfile(eachFile);
+            } catch (HyscaleException e) {
+                return null;
+            }
+        }).collect(Collectors.filtering(each -> {
+            return each.equals(this);
+        }, toSet()));
+
+        boolean multipleProfileOfSameService = false;
+        Set<String> multipleProfilesServices = new HashSet<String>();
+        for (File eachProfile : profileFiles) {
+            String serviceName = ServiceProfileUtil.getServiceNameFromProfile(eachProfile);
+            multipleProfileOfSameService = serviceFromProfiles.add(serviceName);
+            if (!multipleProfileOfSameService) {
+                multipleProfilesServices.add(serviceName);
+            }
+        }
+
+        if (!multipleProfilesServices.isEmpty()) {
+            throw new HyscaleException(ControllerErrorCodes.UNIQUE_PROFILE_REQUIRED, multipleProfilesServices.stream().collect(Collectors.joining(",")));
+        }
+
+        if (strictProfile) {
+            serviceFromSpec.removeAll(serviceFromProfiles);
+            if (!serviceFromSpec.isEmpty()) {
+                throw new HyscaleException(ControllerErrorCodes.SERVICE_NOT_PROVIDED_FOR_PROFILE, serviceFromSpec.stream().collect(Collectors.joining(",")));
```

Comment:
Strict profile implies profile not provided for service right?
Also we need to have Service not provided for profile check

Reply:
Reverse check? How do you validate this case in case of -P. 
You always operate on the service spec. It is just ignored

Reply:
But in case of -p it is required right? We have to prompt the user in case we are ignoring profiles.